src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -92,6 +92,31 @@ protected string BuildRedirectUri(string targetPath)
             return Request.Scheme + "://" + Request.Host + OriginalPathBase + targetPath;
         }
 
+        /// <summary>
+        /// Handles errors by redirecting to an error handler path if configured.  Otherwise sets the status code to 500.
+        /// </summary>
+        /// <param name="error"></param>
+        /// <returns></returns>
+        protected virtual Task HandleErrorAsync(ErrorContext error)
```

Comment:
But actually re the error page, just throw an uncaught exception to trigger the error page.  That's the new pattern for that.  We are nuking all the try/catch alls and using the error middleware to catch all those.